k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -167,36 +167,49 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
    * @return an [[Activity]]`[T]` updated by [[Watch]] events on this object,
    *         where `T` is the return type of the `onEvent` function
    */
-  def activity[T](convert: G => T)(onEvent: (T, W) => T): Activity[T] =
+  def activity[T](
+    convert: G => T,
+    labelSelector: Option[String] = None,
+    fieldSelector: Option[String] = None
+  )(onEvent: (T, W) => T): Activity[T] =
     Activity(Var.async[Activity.State[T]](Activity.Pending) { state =>
       val closeRef = new AtomicReference[Closable](Closable.nop)
-      val pending = get(retryIndefinitely = true)
+      val pending = get(
+        labelSelector = labelSelector,
+        fieldSelector = fieldSelector,
+        retryIndefinitely = true)
         // since we're retrying the GET request forever, this `onFailure`
         // should probably never fire. but who knows?
         .onFailure { e =>
-          log.warning(s"k8s failed to get resource at $path: $e")
-          state.update(Activity.Failed(e))
-        }
+        log.warning(s"k8s failed to get resource at $path: $e")
+        state.update(Activity.Failed(e))
+      }
         // otherwise, update the activity with the initial state, and
         // apply the onEvent function to each successive watch event in
         // the stream.
         .onSuccess { initial =>
-          val initialState = convert(initial)
-          state.update(Activity.Ok(initialState))
+        val initialState = convert(initial)
```

Comment:
this seems like maybe intellij got some weird IDEAs in it's head about formatting (see what I did there?)